### ISSUE TITLE
Link error when using cross-module inlining

### ISSUE THREAD

**Issue by ximion** [ldc-developers/ldc#1727]
I am running LDC on almost-Git-master (the current Git master branch doesn't compile for me, and I haven't found out yet why that happens, maybe another bug)

When compiling a project (that uses concurrency) with LDC, the compile and link steps work flawlessly. But as soon as I do a build with optimizations (`-O` flag passed), I run into the following link error:

```
In function `_D4core6thread5Fiber5resetMFNbZv':
/usr/lib/ldc/x86_64-linux-gnu/include/d/core/thread.d:4375: undefined reference to `_D4core6thread5Fiber5resetMFNbZ9__requireMFNbZv'
```

Any idea on why this happens? (I currently don't have time to run Dustmite on it, but could do that maybe later next week, unless of course the cause for this bug is obvious)

**Comment by kinke** [comment]
Pretty sure it's cross-module inlining - try adding `-enable-cross-module-inlining=false` to the command line. Master should definitely compile and work fine though, we're green on all 3 build systems.

**Comment by ximion** [comment]
Jup, that fixes it :-)

Yeah, since the CI says everything is okay, I want to find the problem in my setup first that is preventing an LDC rebuild.

**Comment by JohanEngelen** [comment]
@ximion Dustmiting is much appreciated.
Which git commit hash are you using?

(btw, with a `-O` build, are you sure you still want contracts enabled? I.e. perhaps you also want to pass `-release`)

**Comment by ximion** [comment]
Commit-ID... Good question, since I already pulled from master and don't remember when I compiled it last.
`ldc2 --version` returns:
```
LDC - the LLVM D compiler (375597):
  based on DMD v2.071.2-b2 and LLVM 3.8.0
```

For this particular project I want to have the debug symbols and contract-safeguards still present - the code is running in production, but I still want to have all the details as soon as something fails. If everything is working without issues, I can do a `release` build instead of a `debugoptimized` build and ship that.
